src/dsql/StmtNodes.cpp

```diff
@@ -647,9 +647,16 @@ const StmtNode* BlockNode::execute(thread_db* tdbb, jrd_req* request, ExeState*
 
 				if (handled && transaction != sysTransaction)
 				{
-					// Check that exception handlers wee executed in context of right savepoint.
+					// Check that exception handlers were executed in context of right savepoint.
 					// If not - mirror copy of CORE-4424 or CORE-4483 is around here.
-					fb_assert(transaction->tra_save_point && transaction->tra_save_point->sav_number == count);
+					// Except the case of nested exception handlers and throwing in inner one.
+					// In this case execution flow is like this:
+					//		inner before (block that rollbacking savepoints above)
+					//		outer before
+					//		outer after  (this block)
+					//		inner after
+					// Because of this following assert is commentd out
+					//fb_assert(transaction->tra_save_point && transaction->tra_save_point->sav_number == count);
```

Comment:
These are parts of code managing savepoints that are placed before and after call of looper for exception handlers, as written in parentheses.